src/codec_aom.c

```diff
@@ -345,12 +380,26 @@ static avifBool avifProcessAOMOptionsPostInit(avifCodec * codec)
         for (int j = 0; aomOptionDefs[j].name; ++j) {
             if (!strcmp(entry->key, aomOptionDefs[j].name)) {
                 match = AVIF_TRUE;
-                int val;
-                avifBool parsed;
-                if (aomOptionDefs[j].enums) {
-                    parsed = aomOptionParseEnum(entry->value, aomOptionDefs[j].enums, &val);
-                } else {
-                    parsed = aomOptionParseInt(entry->value, &val);
+                // store either an int, an unsigned int or a char *.
+                void * val = NULL;
```

Comment:
This is not type-safe, because 'val' will be passed to aom_codec_control() at line 407 below, which expects its third argument to be of a specific type depending on the second argument.

I suggest declaring 'val' to be a local variable of the correct type in each of the cases of the switch statement, and duplicate the aom_codec_control() call in all the cases.